gitlab_sync.go

```diff
@@ -54,10 +54,38 @@ func syncRemoteRef(log *logrus.Entry, org, repo, ref string, conf *config) error
 		return err
 	}
 
+	depths := []int{5, 10, 50, 100, 0} // 0 = infinite depth
+	for depth := range depths {
+		err := fetchCmd(ref, repo, state, depth)
+		if err != nil {
+			break
```

Comment:
would this maybe be:
```suggestion
		if err == nil {
			break
```
?